tests/0122-buffer_cleaning_after_rebalance.c

```diff
@@ -0,0 +1,172 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test.h"
+/* Typical include path would be <librdkafka/rdkafka.h>, but this program
+ * is built from within the librdkafka source tree and thus differs. */
+#include "rdkafka.h"  /* for Kafka driver */
+
+static const int produce_msg_cnt = 400;
+
+typedef struct consumer_s {
+        const char *what;
+        rd_kafka_queue_t *rkq;
+        int timeout_ms;
+        int consume_msg_cnt;
+        rd_kafka_t *rk;
+        uint64_t testid;
+} consumer_t;
+
+static int test_consumer_batch_queue (void *arguments) {
+        struct consumer_s *args = arguments;
+        int msg_cnt = 0;
+        int i;
+        test_timing_t t_cons;
+        test_msgver_t mv;
+
+        rd_kafka_queue_t *rkq = args->rkq;
+        int timeout_ms = args->timeout_ms;
+        const int consume_msg_cnt = args->consume_msg_cnt;
+        rd_kafka_t *rk = args->rk;
+        uint64_t testid = args->testid;
+        const char *what = args->what;
+
+        rd_kafka_message_t **rkmessage = malloc(consume_msg_cnt * sizeof(*rkmessage));
+
+        test_msgver_init(&mv, testid);
+
+        TIMING_START(&t_cons, "CONSUME");
+
+        while ((msg_cnt = rd_kafka_consume_batch_queue(rkq, timeout_ms, rkmessage, consume_msg_cnt)) == 0)
+                continue;
+
+        for (i = 0; i < msg_cnt; i++) {
+                if (test_msgver_add_msg(rk, &mv, rkmessage[i]) == 0)
+                        TEST_FAIL("The message is not from testid %"PRId64" \n", testid);
+        }
+
+        test_msgver_verify(what, &mv, TEST_MSGVER_ORDER|TEST_MSGVER_DUP, 0, produce_msg_cnt/2);
+        test_msgver_clear(&mv);
+
+        for (i = 0; i < consume_msg_cnt; i++)
+                rd_kafka_message_destroy(rkmessage[i]);
+
+        TIMING_STOP(&t_cons);
+        return 0;
+}
+
+
+/**
+ * Consume with batch + queue interface
+ *
+ */
+static void do_test_consume_batch (const char *strategy) {
+        const int partition_cnt = 4;
+        rd_kafka_queue_t *rkq1, *rkq2;
+        const char *topic;
+        rd_kafka_t *c1;
+        rd_kafka_t *c2;
+        int p;
+        const int timeout_ms = 30000;
+        uint64_t testid;
+        const int consume_msg_cnt = 500;
+        rd_kafka_conf_t *conf;
+        struct consumer_s c1_args;
+        struct consumer_s c2_args;
+
+        SUB_TEST("partition.assignment.strategy = %s\n", strategy);
+
+        test_conf_init(&conf, NULL, 60);
+        test_conf_set(conf, "enable.auto.commit", "false");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+
+        testid = test_id_generate();
+
+        /* Produce messages */
+        topic = test_mk_topic_name(__FUNCTION__, 1);
```

Comment:
We'll want the test number in the topic name, so suggest you just do `test_mk_topic_name("0122-buffer_cleaning")